### PR TITLE
Use R 3.2.0 features

### PR DESCRIPTION
**Pull request by MichaelChirico**
Newly-available functions taken from here:

https://github.com/r-lib/lintr/blob/f00f4a9a715346a3e696aab9cffff2756833e50a/R/backport_linter.R#L152-L159

Most significant are `lengths()`, `trimws()`, and `strrep()`.

### CODE REVIEW

File: R/data.table.R [hunk 1]
```diff
@@ -2849,7 +2842,7 @@ setDT = function(x, keep.rownames=FALSE, key=NULL, check.names=FALSE) {
   }
   # check no matrix-like columns, #3760. Other than a single list(matrix) is unambiguous and depended on by some revdeps, #3581
   if (length(x)>1L) {
-    idx = vapply_1i(x, function(xi) length(dim(xi)))>1L
+    idx = cols_with_dims(x)
```
Comment (MichaelChirico) [inline R/data.table.R:2845]: wild that this change is apparently causing some .internal.selfref issue. I've checked `idx` is the same with both approaches. I'm just going to revert this for now.

**Comment by MichaelChirico** [comment]
@jangorecki anything to change here before merging to `1-15-99`?

**Comment by jangorecki** [comment]
In some PRs I recall using 3.1's (vapply's) `lengths()`. It may be better re-run lint/grep after merging PRs in the queue and catching all usage, not just current master.

**Comment by MichaelChirico** [comment]
Still it would be good to turn on 3.2.0 ASAP to allow using the new functions. WDYT about this approach:

 1. This PR simply flips the DESCRIPTION dep
 2. Next PR turns on CI for linting
 3. Then we apply the rest of this PR, i.e. switching to 3.2.0+ "style" internally
 4. Turn on a linter for preventing backsliding

That way, we're "allowed" to use 3.2.0 sooner, but only "enforce" doing so all at once.

**Comment by MichaelChirico** [comment]
@tdhock this is ready for review now that we have linting in CI. #6100 (downstream of this in Graphite stack) has the custom linter referenced to help prevent backsliding.

**Comment by MichaelChirico** [comment]
* **#6100** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/6100?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6099** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/6099?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#5838** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/5838?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MichaelChirico and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by MichaelChirico** [comment]
Thanks for the reviews @tdhock, this PR is also part of the same "stack" -- I can submit them all at once with your review here. :tada:

**Review by tdhock — APPROVED** [review]
looks good thanks

**Comment by MichaelChirico** [comment]
### Merge activity

* **May 4, 12:06 PM EDT**: @MichaelChirico started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/Rdatatable/data.table/5838).

**Pull request merged**